internal/services/recoveryservices/site_recovery_replicated_vm_resource.go

```diff
@@ -108,7 +109,11 @@ func resourceSiteRecoveryReplicatedVM() *pluginsdk.Resource {
 				Optional:         true,
 				ValidateFunc:     azure.ValidateResourceID,
 				DiffSuppressFunc: suppress.CaseDifference,
+				ConflictsWith: []string{
+					"target_zone",
```

Comment:
should target_zone not also be conflicting with this property?

Reply:
Hi @katbyte , thanks for your review. I believe those 2 args are conflicting. Confirmed in the portal that only VMs that are deployed in a zone can have zone to zone replication enabled. And when creating a VM, if a zone is selected, then it cannot be placed into an availability set at the same time. So when doing zone to zone replication, the target VM will be placed in a zone as well, which is conflicting with an availability set, same as when creating a new VM.

Reply:
when an property conflicts with another both properties must have the Conflicts With set. the target zone property does not conflict with this property?

Reply:
Hi @katbyte , I have extracted the schema from commonschema to add the conflictwith set. hope this approach is ok? please let me know if there are any issues. thanks.

Reply:
apologies @cheng514 - seems i was incorrect about it being required on both properties, please sse toms comment below about undoing the last change

Reply:
thanks @katbyte, I saw you have helped with reverting the change, thanks a lot for that!